### DISCUSSION TITLE
High upstream_response_time in NGINX logs, but the actual time to serve the response from the upstream is low.

### DISCUSSION THREAD

**Discussion by anuragiitp** [nginx/nginx#188] [General]
What could cause a high upstream_response_time (55 seconds) when the request completion time in the upstream logs is very low (12 ms)? Multiple requests are facing the same issue intermittently under high load, and it also delays the NGINX health checks.

also when the log line with "referrer:" logged ? for a single request these are getting logged 4-5 times?

{"line":", client: 10.153.61.211, server: , request: \"GET /pcontent/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJtaWQiOiIzNDc4NjE0MTMzIiwidXJsIjoiL3AwYXA2Mmwyb2Jmbi8iLCJleHAiOjE3MjYxNDIwNjR9.gV_BtaCRRa-4qP6WGVlNMe_2UCeqIenwTtoz-PQho9k/p0ap62l2obfn/default/data/resources/ha/9c5d0fdad3504d819a9ac6ec4b4dbcee.png HTTP/1.1\", host: \"xxx.xxxx.com\", referrer: \"https://xxx.xxx.com/pcontent/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJtaWQiOiIzNDc4NjE0MTMzIiwidXJsIjoiL3AwYXA2Mmwyb2Jmbi8iLCJleHAiOjE3MjYxNDIwNjR9.gV_BtaCRRa-4qP6WGVlNMe_2UCeqIenwTtoz-PQho9k/p0ap62l2obfn/default/index.htm?airspeed=1\u0026capi_player=1\u0026isLive=false\u0026ticket=xxxx\u0026capi_player=1\u0026isSourcePresent=true\u0026isSupported=true\u0026userLanguage=en\u0026init_slide=149\"","source":"stderr","tag":"50ad95c68f84","attrs":{"buildVersion":"dev_12.6.0_5","env":"PROD","instanceID":"i-01863d5697c17fd72","region":"us-west-2","service":"cpscahe"}}

**Conf file:**

server {
	listen 127.0.0.1:9000;
	server_name localhost;
	allow 127.0.0.1;
	deny all;
	location = /status {
		stub_status on;
		access_log off;
	}
}

server {
	listen 8443;
	keepalive_timeout 70;

	set $cps_secret "cps_secret_value";

	set_by_lua_block $uuid {
		if ngx.var.http_x_request_id == nil then
		local uv = ngx.var.request_id
		return uv
		else
		return ngx.var.http_x_request_id
		end
	}

  set_by_lua_block $obscured_session {
    local breezesession = ""
    local cookie_value = ngx.var.http_cookie
    if cookie_value then
      local session_value = string.match(cookie_value, "BREEZESESSION=([^;]+)")
      if session_value then
        breezesession = "****" .. string.sub(session_value, -5)
      end
    end
    return breezesession
  }

	add_header X-request-id $uuid always;

	location = /health {
		log_not_found off;
		access_log off;
		content_by_lua_block {
			ngx.status = ngx.HTTP_OK
			ngx.say('OK')
			ngx.flush(true)
		}
	}

	location / {
		return 404;
	}

	location ~ ^/api/json {
		internal;
		proxy_no_cache on;
		proxy_pass https://$host;
	}

	location ~ ^/pcontent/mtoken/validate {
		access_by_lua_block {
			local cacheHelper = require "lcc.cacheHelper"
			cacheHelper.setMTokenAsCookie()
		}
		content_by_lua_block {
			ngx.say('OK')
			ngx.flush(true)
		}
	}

	location ~ ^/common/ {
		# Common settings
		include upstreams/proxy.settings;
		include upstreams/cors.settings;

		proxy_cache common;
		proxy_cache_valid 200 206 1440m;

		proxy_cache_lock_timeout 5s;

		proxy_cache_lock_age 30s;

		more_clear_headers 'Cache-Control';
		more_clear_headers 'Set-Cookie';
		add_header Cache-Control $cache_control_header;

		proxy_pass https://$host;

	}

	location ~ ^/pcontent/(?<c_token>[^/]+)/ {
		# Common settings
		include upstreams/proxy.settings;
		include upstreams/cors.settings;

		proxy_cache pcontent;
		proxy_cache_valid 200 206 320m;

		proxy_cache_lock_timeout 0s;

		proxy_cache_lock_age 30s;

		set $set_m_token "";
		set $newuri "";
		
		access_by_lua_block {
			local cacheHelper = require "lcc.cacheHelper"
			cacheHelper.processContentRequest()
		}

		rewrite_by_lua_block {
			local url = ngx.var.request_uri

			-- the built in rewrite function doesn't handle encoding properly, so, using string substitution to remove "/pcontet/<<c_token>>" from the url before sending to xxx
			local rewritten_url = url:gsub("^/pcontent/([^/]+)/(.*)$", "/%2")
			ngx.var.newuri = ngx.var.host .. rewritten_url

			-- still update the uri to get consistent logs
			ngx.req.set_uri(rewritten_url)
		}
		
		header_filter_by_lua_block {
			ngx.header["Set-Cookie"] =nil
			ngx.header["Cache-Control"] =ngx.var.cache_control_header
			if ngx.var.set_m_token ~= nil then
				ngx.header["Set-Cookie"] = ngx.var.set_m_token;
			end
		}

		proxy_pass https://$newuri;
	}

}

**Comment by Maryna-f5** [comment]
This discussion has been inactive for a long time, closing it.
